From 406f37c546786149ccffd6f4b6ed3a71eba77d3c Mon Sep 17 00:00:00 2001
From: Wenpeng Liang <liangwenpeng@huawei.com>
Date: Sat, 8 Jan 2022 18:52:17 +0800
Subject: [PATCH] RDMA/hns: Enable the cmd mask of uverbs to create and destroy
 AH
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 406f37c546786149ccffd6f4b6ed3a71eba77d3c


driver inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4NI61

--------------------------------------------------------------

If the corresponding bit is not set, the user will not be able to
create AH.

Fixes: 9a4435375cd1 ("IB/hns: Add driver files for hns RoCE driver")
Signed-off-by: Wenpeng Liang <liangwenpeng@huawei.com>
sigend-off-by: Guofeng Yue <yueguofeng@hisilicon.com>

Reviewed-by: Yangyang Li <liyangyang20@huawei.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/infiniband/hw/hns/hns_roce_main.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/infiniband/hw/hns/hns_roce_main.c
+++ b/drivers/infiniband/hw/hns/hns_roce_main.c
@@ -598,7 +598,9 @@ static int hns_roce_register_device(stru
 		(1ULL << IB_USER_VERBS_CMD_CREATE_QP) |
 		(1ULL << IB_USER_VERBS_CMD_MODIFY_QP) |
 		(1ULL << IB_USER_VERBS_CMD_QUERY_QP) |
-		(1ULL << IB_USER_VERBS_CMD_DESTROY_QP);
+		(1ULL << IB_USER_VERBS_CMD_DESTROY_QP) |
+		(1ULL << IB_USER_VERBS_CMD_CREATE_AH) |
+		(1ULL << IB_USER_VERBS_CMD_DESTROY_AH);
 
 	if (hr_dev->caps.flags & HNS_ROCE_CAP_FLAG_REREG_MR) {
 		ib_dev->uverbs_cmd_mask |= (1ULL << IB_USER_VERBS_CMD_REREG_MR);
